c7n/resources/timestream.py

```diff
@@ -144,6 +145,42 @@ class TimestreamInfluxDBSubnetFilter(SubnetFilter):
     RelatedIdsExpression = "vpcSubnetIds[]"
 
 
+@TimestreamInfluxDB.filter_registry.register('db-parameter')
+class Parameter(ValueFilter):
+    """Filter timestream influxdb instances based on parameter values.
+
+    :example:
+
+    .. code-block:: yaml
+
+       policies:
+         - name: filter-timestream-influxdb-instance
+           resource: aws.timestream-influxdb
+           filters:
+            - type: db-parameter
+              key: fluxLogEnabled
+              value: True
+    """
+    permissions = ('timestream-influxdb:GetDbParameterGroup',)
+    schema = type_schema('db-parameter', rinherit=ValueFilter.schema)
+    annotation_key = 'c7n:MatchedDBParameter'
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('timestream-influxdb')
+        results = []
+        for r in resources:
+            if self.annotation_key not in r and \
+                'dbParameterGroupIdentifier' in r:
+                r[self.annotation_key] = client.get_db_parameter_group(
```

Comment:
this should keep a map of parameter groups by identifier, as most of them will be common across servers rather needing to refetch per server. ie 100 dbs, 5 parameter groups, this could be reduced to fetch only once per parameter group vs once per db.

Reply:
Got it. Thanks for the suggestion! I updated my PR.